port/unix/omrsysinfo.c

```diff
@@ -4526,8 +4526,13 @@ omrsysinfo_cgroup_subsystem_iterator_next(struct OMRPortLibrary *portLibrary, st
 				goto _end;
 			}
 	  	} while ((part = strtok_r(NULL, "\n", &saveptr)) != NULL);
-	} else {
-		portLibrary->mem_free_memory(portLibrary, state->fileContent);
+		state->multiLineCounter = state->multiLineCounter + 1;
```

Comment:
> Shouldn't the increment happen unconditionally in the loop?

if the token matches `currentElement->metricKeyInFile` then the multiLineCounter is incremented and goes to the `_end` block of the function. In case if the metric is missing from that file then as `currentElement` is dependent on multiLineCounter (`currentElement += state->multiLineCounter;`) it always end up in the same metric to search again and again as the condition `if (state->multiLineCounter < subsystemMetricMapElement->metricElementsCount)` will be satisfied always. So if the metric is not found after parsing the whole cached file content i'm incrementing the `state->multiLineCounter` outside the loop. 

Please correct me if i'm wrong

Thanks